ptp/generate.go

```diff
@@ -231,6 +318,48 @@ func generateToolpath(argv []string) error {
 					}
 				}
 			}
+		} else if line.IsArcMove() {
+			x, y, z := writer.GetCurrentPosition()
+			if lineX, ok := line.Params["x"]; ok {
+				x = lineX
+			}
+			if lineY, ok := line.Params["y"]; ok {
+				y = lineY
+			}
+			if lineZ, ok := line.Params["z"]; ok {
+				z = lineZ
+			}
+
+			i := float32(0)
+			j := float32(0)
+			if lineI, ok := line.Params["i"]; ok {
+				i = lineI
+			}
+			if lineJ, ok := line.Params["j"]; ok {
+				j = lineJ
+			}
+
+			currentX, currentY, _ := writer.GetCurrentPosition()
+			clockwise := line.Command == "G2"
+			segments := getArcMoveSegments(clockwise, currentX, currentY, x, y, i, j, z)
+			if !seen {
+				seen = true
+			}
+			if state.transitioning {
+				// TODO: Is this needed? transitions should never be arcs?
```

Comment:
I think this isn't technically needed but also seems like it doesn't hurt to have – who knows, maybe we generate circular transition towers in the future?